crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -280,85 +255,131 @@ where
+                        Ok(success) => {
+                            debug!(
+                                ?tx_digest,
+                                ?conflicting_tx_digest_to_retry,
+                                "Retried conflicting transaction success: {}",
+                                success
+                            );
+                            if success {
+                                self.metrics
+                                    .total_successful_attempts_retrying_conflicting_transaction
+                                    .inc();
+                            }
+                            let err = Err(Some(QuorumDriverError::ObjectsDoubleUsed {
                                 conflicting_txes: conflicting_tx_digests,
-                                retried_tx_digest: Some(retried_tx_digest),
+                                retried_tx: Some(conflicting_tx_digest),
                                 retried_tx_success: Some(success),
-                            },
-                        );
+                            }));
+                            return err;
+                        }
                     }
+                } else {
+                    // If no retryable conflicting transaction was returned that means we have >= 2f+1 good stake for
+                    // the original transaction + retryable stake. Will continue to retry the original transaction.
+                    debug!(
+                        ?errors,
+                        "When getting tx cert, observed {} conflicting transactions. Tx {tx_digest:} is in retryable state. Conflicting Txes: {conflicting_tx_digests:?}", 
+                        conflicting_tx_digests.len()
+                    );
+                    return Err(None);
                 }
-                Err(
-                    QuorumDriverInternalError::TransactionErrorWithConflictingTransactions {
-                        conflicting_txes: conflicting_tx_digests,
-                        retried_tx_digest: None,
-                        retried_tx_success: None,
-                    },
-                )
+                let err = Err(Some(QuorumDriverError::ObjectsDoubleUsed {
+                    conflicting_txes: conflicting_tx_digests,
+                    retried_tx: None,
+                    retried_tx_success: None,
+                }));
+                debug!(
+                    ?tx_digest,
+                    "Non retryable error when getting tx cert: {err:?}"
+                );
+                err
+            }
+
+            Err(AggregatorProcessTransactionError::FatalConflictingTransaction {
+                errors,
+                conflicting_tx_digests,
+            }) => {
+                debug!(
+                    ?errors,
+                    "When getting tx cert, observed {} conflicting transactions. Tx {tx_digest:} is in nonretryable state. Conflicting Txes: {conflicting_tx_digests:?}",
```

Comment:
nit
```suggestion
                    "Observed Tx {tx_digest:} fully equivocated. Conflicting Txes: {conflicting_tx_digests:?}",
```

Reply:
Think Lefteris suggested not using "equivocation" when it is triggered by the client so I will say "double spend attempted", but your suggestion is accepted